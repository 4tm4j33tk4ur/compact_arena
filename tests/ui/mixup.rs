extern crate compact_arena;

use compact_arena::in_nano_arena;

fn main() {
    in_nano_arena!(a, {
        in_nano_arena!(b, {
            let _ = a.add(1usize);
            let x = a.add(1usize);
            let y = b.add(1usize);
            let _ = b[x];
        })
    });
}
