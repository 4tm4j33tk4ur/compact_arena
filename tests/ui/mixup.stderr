error[E0597]: `tag` does not live long enough
  --> $DIR/mixup.rs:7:9
   |
6  |   /     in_nano_arena!(a, {
7  |   |         in_nano_arena!(b, {
   |  _|_________^
8  | | |             let _ = a.add(1usize);
9  | | |             let x = a.add(1usize);
10 | | |             let y = b.add(1usize);
11 | | |             let _ = b[x];
12 | | |         })
   | | |          ^
   | | |          |
   | |_|__________borrowed value does not live long enough
   |   |          `tag` dropped here while still borrowed
13 |   |     });
   |   |_______- borrow might be used here, when `arena` is dropped and runs the `Drop` code for type `compact_arena::NanoArena`
   |
   = note: values in a scope are dropped in the opposite order they are defined
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: aborting due to previous error

For more information about this error, try `rustc --explain E0597`.
